metadata-ingestion/src/datahub/utilities/file_backed_collections.py

```diff
@@ -118,19 +137,18 @@ class FileBackedDict(MutableMapping[str, _VT], Generic[_VT]):
     This class is not thread-safe.
     """
 
-    filename: pathlib.Path
-    tablename: str = field(default=_DEFAULT_TABLE_NAME)
+    # Use a predefined connection, able to be shared across multiple FileBacked* objects
+    connection: Optional[ConnectionWrapper] = None
+    tablename: str = _DEFAULT_TABLE_NAME
 
-    serializer: Callable[[_VT], SqliteValue] = field(default=_default_serializer)
```

Comment:
i think we need to keep the field() stuff because this is a dataclass

Reply:
I don't think so. We do this in other places, e.g. in https://github.com/datahub-project/datahub/blob/master/metadata-ingestion/src/datahub/ingestion/source/sql/sql_generic.py. https://www.infoworld.com/article/3563878/how-to-use-python-dataclasses.html#toc-3 goes into more detail -- you can hold off using `field` for the most basic cases

Reply:
Yup you're right - I misremembered it